### PR TITLE
Python audit plugin

### PR DESCRIPTION
**Pull request by manner82**
Add support for
- writing audit plugins in python
- returning errstr from other plugins to the audit plugin

### CODE REVIEW

File: plugins/python/regress/check_python_examples.c [hunk 1]
```diff
@@ -861,10 +1010,284 @@ check_python_plugins_do_not_affect_each_other(void)
...
+
+    str_array_free(&data.plugin_argv);
+    data.plugin_argv = create_str_array(3, "id", "--help", NULL);
+
+    VERIFY_INT(python_audit->accept("accepter plugin name", SUDO_POLICY_PLUGIN,
+                                    data.command_info, data.plugin_argv,
+                                    data.user_env, &errstr), SUDO_RC_OK);
+    VERIFY_PTR(errstr, NULL);
+
+    python_audit->close(SUDO_PLUGIN_WAIT_STATUS, 2);  // process exited with 2
```
Comment (millert) [inline plugins/python/regress/check_python_examples.c]: wait status is not the same as exit value.  You can use the W_EXITCODE macro to create a wait status from an exit value and a signal.  For example:
```
python_audit->close(SUDO_PLUGIN_WAIT_STATUS, W_EXITCODE(2, 0));  // process exited with 2
```

File: plugins/python/regress/check_python_examples.c [hunk 2]
```diff
@@ -861,10 +1010,284 @@ check_python_plugins_do_not_affect_each_other(void)
...
+                                    data.user_env, &errstr), SUDO_RC_OK);
+    VERIFY_PTR(errstr, NULL);
+
+    VERIFY_INT(python_audit2->accept("accepter plugin name", SUDO_POLICY_PLUGIN,
+                                    data.command_info, data.plugin_argv,
+                                    data.user_env, &errstr), SUDO_RC_OK);
+    VERIFY_PTR(errstr, NULL);
+
+    python_audit->close(SUDO_PLUGIN_WAIT_STATUS, 2);  // process exited with 2
+    python_audit2->close(SUDO_PLUGIN_WAIT_STATUS, 2);  // process exited with 2
```
Comment (millert) [inline plugins/python/regress/check_python_examples.c]: Should use W_EXITCODE here too

File: plugins/python/example_audit_plugin.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+    def error(self, plugin_name, plugin_type, audit_msg, command_info) -> int:
+        self._log("Plugin {} (type={}) got an error: {}".format(
+            plugin_name, self.__plugin_type_str(plugin_type), audit_msg))
+
+    def close(self, status_kind: int, status: int) -> None:
+        if status_kind == sudo.EXIT_REASON.NO_STATUS:
+            self._log("The command was not executed")
+        elif status_kind == sudo.EXIT_REASON.WAIT_STATUS:
+            self._log("Command returned with exit code {}".format(status))
```
Comment (millert) [inline plugins/python/example_audit_plugin.py]: wait status != exit code.  You probably want to do the equivalent of the macros in sys/wait.h to determine whether the command exited normally or was killed by a signal.

File: plugins/python/regress/check_python_examples.c (hunk 1 above)
Comment (manner82) [reply plugins/python/regress/check_python_examples.c]: I have fixed these. Replaced one of the invocation with signal termination, the other with exit code termination, so both are tested.

**Comment by manner82** [comment]
Also updated the man pages for the error string returning and audit plugin.

**Pull request merged**